dandischema/models.py

```diff
@@ -58,8 +64,12 @@
 )
 ASSET_UUID_PATTERN = r"^dandiasset:" + UUID_PATTERN
 VERSION_PATTERN = r"\d{6}/\d+\.\d+\.\d+"
-DANDI_DOI_PATTERN = rf"^10.(48324|80507)/dandi\.{VERSION_PATTERN}"
-DANDI_PUBID_PATTERN = rf"^DANDI:{VERSION_PATTERN}"
+DANDI_DOI_PATTERN = (
+    rf"^10\.{DATACITE_DOI_ID_PATTERN}/{ID_PATTERN.lower()}\.{VERSION_PATTERN}"
+)
```

Comment:
for @asmacdo and 
- https://github.com/dandi/dandi-archive/pull/2012
we might right away mandate either "empty" or "any" DOI, ie.

```suggestion
DANDI_DOI_PATTERN = (
    rf"^10\.{DATACITE_DOI_ID_PATTERN}/{ID_PATTERN.lower()}\.{VERSION_PATTERN}"
    if DATACITE_DOI_ID_PATTERN else "^$" # or ".*" for "any"
)
```

but we need to decide which one in that design PR.  Might be easier when implemented to see the manifestations

Reply:
I still think that we should do above suggestion so we allow for instances which are not connected to dandicite (e.g. local ones), but still should be allowed to "publish" versions of dandisets on them.

Reply:
This done, as discussed in the meeting, the `doi` field in `PublishedDandiset` is defined conditionally.